src/common/common-types.ts

```diff
@@ -209,9 +211,19 @@ export interface NodeData {
     readonly percentComplete?: number;
     readonly minWidth?: number;
     readonly minHeight?: number;
+    readonly collidingEdge?: Edge<EdgeData>;
 }
 export interface EdgeData {
     readonly complete?: boolean;
+    sourceX?: number;
+    sourceY?: number;
+    targetX?: number;
+    targetY?: number;
+    type?: NonNeverType;
+    edgePath?: string;
+    edgeCenterX?: number;
+    edgeCenterY?: number;
+    colliding?: boolean;
```

Comment:
I am against this change for node and edge data. I can already see a Ctrl+Z completely breaking this.

- `NodeData.collidingEdge` is just like `hoveredNode` and can be a simple `useState` in global. Same for `EdgeData.colliding`.
- `type` can be derived from `typeState` and doesn't need to be stored.
- `{source,target}{X,Y}` are already props of `Edge<T>`.
- `edgeCenter{X,Y}` and `edgePath` can be calculated on the fly and don't need to be stored.

Reply:
What about Edith edgePath? We'd need to recalculate that

Reply:
Also take a look at this: https://discord.com/channels/771389069270712320/859774873500778517/1041561117471805490

Reply:
> I can already see a Ctrl+Z completely breaking this

When you Ctrl+Z and an edge changes it would update this

Reply:
> What about edgePath? We'd need to recalculate that

Yeah, just recalculate and cache it.

Reply:
If we're gonna recalculate it then I'm going to just copy how it gets each number from the RF code so we don't have to parse it from the string it gives. Could lead to desync in the future but should be a bit more efficient i guess